### PR TITLE
Replace pusher debug with dry-run option

### PR DESCRIPTION
**Pull request by einarmo**
This is a bit more general, the pusher debug option was always a bit artificial. This is intended as a proper way to prevent the extractor from persisting anything, while still letting users see what the extractor would do.

This should mostly not access destinations at all. The exception is if CDF is configured as source, which is intentional.

### CODE REVIEW

**Review by ozangoktan — COMMENTED** [review]
I noticed that Failure buffer would [write to file](https://github.com/cognitedata/opcua-extractor-net/blob/17f97ecc03a70d6da48344ad18b1e40592763542/Extractor/FailureBuffer.cs#L234) regardless of this setting, is this intended?

**Comment by ozangoktan** [comment]
>  This is intended as a proper way to prevent the extractor from persisting anything, while still letting users see what the extractor would do.

https://github.com/cognitedata/opcua-extractor-net/blob/dry-run-mode/Extractor/PubSub/PubSubManager.cs#L97 also seems to persist a config file, though I am not sure dry-run would apply to config files.

File: Extractor/Pushers/InfluxPusher.cs [hunk 1]
```diff
@@ -189,8 +197,13 @@ public InfluxPusher(ILogger<InfluxPusher> log, InfluxPusherConfig config)
 
             if (count == 0) return null;
 
+            if (fullConfig.DryRun)
+            {
+                log.LogInformation("Dry run enabled. Would insert {Count} events to influxdb", count);
+                return null;
+            }
+
             var points = evts.Select(UAEventToInflux);
```
Comment (ozangoktan) [inline Extractor/Pushers/InfluxPusher.cs:206]: Is there a reason to exclude this conversion from the dry-run?

**Review by ozangoktan — APPROVED** [review]
Looks good, left a couple comments that would be nice to look into before merging.

File: Extractor/Pushers/InfluxPusher.cs [hunk 2]
```diff
@@ -120,14 +122,20 @@ public InfluxPusher(ILogger<InfluxPusher> log, InfluxPusherConfig config)
 
             var ipoints = new List<IInfluxDatapoint>();
 
+            int groupCount = 0;
```
[the remark this answers is not in the document]
Comment (einarmo) [reply Extractor/Pushers/InfluxPusher.cs:125]: It is? Line 136.

File: Extractor/Pushers/InfluxPusher.cs (hunk 1 above)
Comment (einarmo) [reply Extractor/Pushers/InfluxPusher.cs:206]: Doesn't look like it, I'll include it.

File: Extractor/Pushers/InfluxPusher.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ozangoktan) [reply Extractor/Pushers/InfluxPusher.cs:125]: Ah yes, it is there. Must have missed it.

**Pull request merged**